dafni_cli/api/session.py

```diff
@@ -327,15 +348,88 @@ def _authenticated_request(
 
         return response
 
+    def get_error_message(self, response: requests.Response) -> Optional[str]:
+        """Attempts to find an error message from a failed request response
+
+        Args:
+            response (requests.Response): The failed request response
+
+        Returns:
+            Optional[str]: String representing an error message or None
+                           if none was found
+        """
+
+        # Try and get JSON data from the response
+        try:
+            error_message = None
+            decoded_response = response.json()
+            # Some requests have an error and error_message, in which case we
+            # want to override with the latter
+            if "error" in decoded_response:
+                error_message = f"Error: {decoded_response['error']}"
+            if "error_message" in decoded_response:
+                error_message = f"{error_message}, {decoded_response['error_message']}"
+            elif "errors" in decoded_response:
+                error_message = "The following errors were returned:"
+                for error in decoded_response["errors"]:
+                    error_message += f"\nError: {error}"
+            # Special case when uploading dataset metadata that's invalid
+            elif "metadata" in decoded_response:
```

Comment:
I think this is actually a bug. The errors should be returned in an "errors" field not in  "metadata" - I'll add a task to JIRA to address this. Good spot